### PR TITLE
introduce --cluster-monitor flag to control cluster health

### PR DESCRIPTION
**Pull request by snowplayfire**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
introduce --cluster-monitor flag to control whether monitor cluster health or not
```

### CODE REVIEW

**Comment by snowplayfire** [comment]
@rainbowmango

File: pkg/controllers/cluster/cluster_controller.go [hunk 1]
```diff
@@ -103,10 +104,13 @@ func (c *Controller) Start(ctx context.Context) error {
...
 func (c *Controller) SetupWithManager(mgr controllerruntime.Manager) error {
-	return utilerrors.NewAggregate([]error{
+	errors := []error{
 		controllerruntime.NewControllerManagedBy(mgr).For(&clusterv1alpha1.Cluster{}).Complete(c),
-		mgr.Add(c),
-	})
+	}
+	if c.ClusterMonitor {
+		errors = append(errors, mgr.Add(c))
+	}
```
Comment (RainbowMango) [inline pkg/controllers/cluster/cluster_controller.go:112]: If `--cluster-monitor==false`, then the `cluster` controller here will not added to `manager`, it equal to `--controllers=-cluster`. Am I right?

Comment (snowplayfire) [reply pkg/controllers/cluster/cluster_controller.go:112]: Not exactly the same. I need  both `--controllers=cluster`  with `--cluster-monitor==false`.

Comment (RainbowMango) [reply pkg/controllers/cluster/cluster_controller.go:112]: Oh, you are right. if `--cluster-monitor==false`, only stopp the `Start` function(monitor).

**Comment by snowplayfire** [comment]
close it.

**Pull request closed without merging**